### DISCUSSION TITLE
Viability for real-time SPA

### DISCUSSION THREAD

**Discussion by ccgauvin94** [nextpyp/.github#78] [Q&A] [answered]
Dear nextPYP team,

First of all, I just wanted to compliment you all on what really looks like a fantastic, well-designed piece of software. 

I'm interested in nextPYP for doing real-time pre-processing and then handing off to Relion for 2D/3D. I've played around a bit with it, and while the real-time pre-processing works exceptionally well, it doesn't seem to be able to export coordinates in real-time. I seem to have to wait for the pre-processing to finish before I can export coordinates. 

I wanted to check if this is correct, or if maybe I'm missing something. Ideally, I'd like to be able to do pre-processing as data is collected in nextPYP, and then pass the particle coordinates (or even CTF corrected micrographs) off to Relion as they are processed/picked, for Relion schemes to import and further process. Is anything like that currently possible? 

Thank you for your time. 

Colin Gauvin
Cryo-EM Facility Manager
Montana State University

**Comment by yezhou867** [comment, marked as answer]
Hi, real-time preprocessing could do a weak export (only aligned images) to RELION format.  It's possible to export coordinates but we haven't enabled that function yet. We can test it a little bit and enable it in the new version. 

<img width="744" alt="Screenshot 2024-03-07 at 3 22 07 PM" src="https://github.com/nextpyp/.github/assets/139471026/8fdf156b-919a-430d-9a86-66e1b23ab92b">

**Reply by ccgauvin94** [reply]
Awesome - I will keep an eye out.

**Reply by abartesaghi** [reply]
This was addressed in version [0.6.4](https://nextpyp.app/files/pyp/0.6.4/docs/index.html).
